assets/Product.wxs

```diff
@@ -73,6 +73,8 @@
                   Return="ignore"
                   Impersonate="no" />
     <InstallExecuteSequence>
+      <!-- Do not remove shortcuts on upgrade to preserve user shortcuts pinned to Taskbar, see https://stackoverflow.com/a/33402698/1810304 -->
+      <RemoveShortcuts>Installed AND NOT UPGRADINGPRODUCTCODE</RemoveShortcuts>
```

Comment:
This will always be true because of a change to force a full install every time to solve other issues:
https://github.com/PowerShell/PowerShell/blob/8dcd5a2e1cb9f92355854948b63ff75fcd871be5/assets/wix/Product.wxs#L41

Reply:
https://github.com/PowerShell/PowerShell/pull/12792